src/build/arg/mod.rs

```diff
@@ -5123,19 +5123,19 @@ where
         arg.val_delim.expect(INTERNAL_ERROR_MSG)
     } else {
         ' '
-    };
+    }
+    .to_string();
     if !arg.val_names.is_empty() {
         // If have val_name.
         match (arg.val_names.len(), arg.num_vals) {
             (1, Some(num_vals)) => {
                 // If single value name with multiple num_of_vals, display all
                 // the values with the single value name.
                 let arg_name = format!("<{}>", arg.val_names.get(0).unwrap());
-                let mut it = iter::repeat(arg_name).take(num_vals).peekable();
-                while let Some(arg_name) = it.next() {
+                for n in 1..=num_vals {
                     write(&arg_name, true)?;
-                    if it.peek().is_some() {
-                        write(&delim.to_string(), false)?;
+                    if n != num_vals {
+                        write(&delim, false)?;
                     }
                 }
             }
```

Comment:
Would it be simpler to do
```rust
write(&arg_name, true)?;
for _ in 1..num_vals {
    write(&delim, false)?;
    write(&arg_name, true)?;
}
```

No idea the impact on code size

Reply:
In at least one of the cases, we've ensured there is at least 1 value.  The second case, I'm less sure about.